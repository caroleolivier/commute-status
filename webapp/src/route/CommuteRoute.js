import React, { Component } from 'react';
import PropTypes from 'prop-types';

import BusStationContainer from './bus/BusStationContainer';
import TubeStationContainer from './tube/TubeStationContainer';

const CommuteRouteStopType = {
    BUS: {
        getComponent: stopConfig => ({
            key: `bus${stopConfig.stationId}`,
            component: <BusStationContainer config={stopConfig} />
        })
    },
    TUBE: {
        getComponent: stopConfig => ({
            key: `tube${stopConfig.stationId}`,
            component: <TubeStationContainer config={stopConfig} />
        })
    }
};

class CommuteRoute extends Component {
    render() {
        const stops = this.props.stops.map(stopConfig =>
            CommuteRouteStopType[stopConfig.type.toUpperCase()].getComponent(stopConfig));
        return (
            <div>
                <h2>{this.props.routeName}</h2>
                <div className="horizontalContainer">
                    {stops.map(stop => <div key={stop.key}>{stop.component}</div>)}
                </div>
            </div>
        );
    }
}

CommuteRoute.propTypes = {
    routeName: PropTypes.string.isRequired,
    stops: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default CommuteRoute;
