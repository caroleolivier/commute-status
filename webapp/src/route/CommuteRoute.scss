.commuteRoute {
    padding: 10px;
}