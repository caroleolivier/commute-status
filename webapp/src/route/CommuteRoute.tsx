import * as React from 'react';

import { StationContainer } from './station/StationContainer';
import * as styles from './CommuteRoute.scss';

export interface ICommuteRouteConfig {
    routeName: string;
    stops: any[];
}

interface ICommuteRouteProps {
    config: ICommuteRouteConfig;
}

export class CommuteRoute extends React.Component<ICommuteRouteProps, {}> {
    render(): JSX.Element {
        return (
            <div className={styles.commuteRoute}>
                <h2>{this.props.config.routeName}</h2>
                <div className="horizontalContainer">
                    {this.props.config.stops.map(stopConfig => <StationContainer key={stopConfig.stationId} config={stopConfig} />)}
                </div>
            </div>
        );
    }
}
