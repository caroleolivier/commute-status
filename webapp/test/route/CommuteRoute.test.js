/* eslint-env browser, jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CommuteRoute from '../../src/route/CommuteRoute';

describe('CommuteRoute', () => {
    describe('Given route name and a list of stops', () => {
        const routeName = 'Via the Moon';
        const stops = [
            {
                type: 'bus',
                stationName: 'River',
                stationId: '490008430E',
                direction: 'Saturne',
                directionId: 'outbound',
                lines: ['L1', 'L2']
            },
            {
                type: 'tube',
                stationName: 'Sea',
                stationId: '940GZZLUTBY',
                direction: 'Mars',
                directionId: 'inbound',
                lines: ['N1']
            }
        ];
        let commuteRouteWrapper;

        beforeEach(() => {
            commuteRouteWrapper = shallow(<CommuteRoute routeName={routeName} stops={stops} />);
        });

        test('it creates the correct stop components', () => {
            const tree = toJson(commuteRouteWrapper);
            expect(tree).toMatchSnapshot('routes');
        });
    });
});
