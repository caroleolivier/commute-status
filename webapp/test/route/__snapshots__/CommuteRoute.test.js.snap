// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes 1`] = `
<div>
  <h2>
    Via the Moon
  </h2>
  <div
    className="horizontalContainer"
  >
    <div
      key="bus490008430E"
    >
      <BusStationContainer
        config={
          Object {
            "direction": "Saturne",
            "directionId": "outbound",
            "lines": Array [
              "L1",
              "L2",
            ],
            "stationId": "490008430E",
            "stationName": "River",
            "type": "bus",
          }
        }
      />
    </div>
    <div
      key="tube940GZZLUTBY"
    >
      <TubeStationContainer
        config={
          Object {
            "direction": "Mars",
            "directionId": "inbound",
            "lines": Array [
              "N1",
            ],
            "stationId": "940GZZLUTBY",
            "stationName": "Sea",
            "type": "tube",
          }
        }
      />
    </div>
  </div>
</div>
`;
